### PR TITLE
ability to specify environment variable options as cli options

### PR DESCRIPTION
**Pull request by basz**
suggested in https://github.com/proophsoftware/prooph-cli/issues/11

### CODE REVIEW

**Comment by sandrokeil** [comment]
@basz Thank you very much for this PR. I'm not sure if it is needed if #16 is merged.

**Comment by codeliner** [comment]
ok @basz @sandrokeil what do we want to do here? Maybe it is a good idea to allow all three: 
1. setting options via input options
2. setting options via env variables 
3. provide Psr4Info via container

maybe: 1. overrides 2. overrides 3. ?

@basz If we go this route you need to add a check before calling f.e. `$classInfo->setSourceFolder($input->getOption('source-folder'));`. Otherwise it would set source folder to NULL if input option is not provided by the user.

File: src/Console/Command/GenerateAggregate.php [hunk 1]
```diff
@@ -84,6 +84,23 @@ protected function configure()
...
+            ->addOption(
+                'source-folder',
+                null,
+                InputArgument::OPTIONAL,
+                'Absolute path to the source folder.'
+            )
+            ->addOption(
+                'package-prefix',
+                null,
+                InputArgument::OPTIONAL,
```
Comment (sandrokeil) [inline src/Console/Command/GenerateAggregate.php]: Please use `InputOption::VALUE_OPTIONAL` instead of `InputArgument::OPTIONAL`

**Comment by sandrokeil** [comment]
We can allow all 3 variants, but the order (last overrides previous) should be:
1. env (lowest)
2. Container (overrides env, because Application is more important than env)
3. argument (overrides previous, the user want this :smile: )

File: src/Console/Command/AbstractGenerateCommand.php [hunk 2]
```diff
@@ -29,6 +29,9 @@ protected function generateClass(InputInterface $input, OutputInterface $output,
 
         /* @var $classInfo ClassInfo */
         $classInfo = $this->getHelper(ClassInfo::class);
+        $classInfo->setSourceFolder($input->getOption('source-folder'));
```
Comment (codeliner) [inline src/Console/Command/AbstractGenerateCommand.php]: Only set if option was specified by user

File: src/Console/Command/GenerateAll.php [hunk 3]
```diff
@@ -108,6 +126,12 @@ protected function execute(InputInterface $input, OutputInterface $output)
                     . $commands['aggregate'];
             }
 
+            /* @var $classInfo ClassInfo */
+            $classInfo = $this->getHelper(ClassInfo::class);
+            $classInfo->setSourceFolder($input->getOption('source-folder'));
```
Comment (codeliner) [inline src/Console/Command/GenerateAll.php]: Only set if option was specified by user

**Comment by codeliner** [comment]
@basz sorry, I did not get a note that you've added new commits. PR is ready now, isn't it?
@sandrokeil Can you take a last look and merge if changes are ok for you?

**Comment by basz** [comment]
yep i was done

**Comment by sandrokeil** [comment]
Yes I will do.

**Pull request merged**